src/clients/api/queries/getXvsBridgeMintStatus/useGetXvsBridgeMintStatus.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback } from 'react';
+import { QueryObserverOptions, useQuery } from 'react-query';
+
+import { GetXvsMintStatusOutput, getXvsBridgeMintStatus } from 'clients/api';
+import FunctionKey from 'constants/functionKey';
+import {
+  getXVSProxyOFTDestContractAddress,
+  getXvsTokenMultichainContract,
+} from 'packages/contracts';
+import { useProvider } from 'packages/wallet';
+import { ChainId } from 'types';
+import { generatePseudoRandomRefetchInterval } from 'utilities';
+
+interface UseGetXvsBridgeFeeInput {
+  destinationChain: ChainId;
+}
+
+export type UseGetTokenBalancesQueryKey = [
+  FunctionKey.GET_XVS_BRIDGE_MINT_STATUS,
+  UseGetXvsBridgeFeeInput,
+];
+
+type Options = QueryObserverOptions<
+  GetXvsMintStatusOutput | undefined,
+  Error,
+  GetXvsMintStatusOutput | undefined,
+  GetXvsMintStatusOutput | undefined,
+  UseGetTokenBalancesQueryKey
+>;
+
+const refetchInterval = generatePseudoRandomRefetchInterval();
+
+const useGetXvsBridgeFeeEstimation = (
+  { destinationChain }: UseGetXvsBridgeFeeInput,
+  options?: Options,
+) => {
+  const { provider } = useProvider({ chainId: destinationChain });
+  const xvsTokenMultichainContract = getXvsTokenMultichainContract({
+    chainId: destinationChain,
+    signerOrProvider: provider,
+  });
+  const chainXvsProxyOftDestContractAddress = getXVSProxyOFTDestContractAddress({
+    chainId: destinationChain,
+  });
+  const getXvsBridgeMintStatusIfAvailable = useCallback(() => {
+    if (xvsTokenMultichainContract && chainXvsProxyOftDestContractAddress) {
+      return getXvsBridgeMintStatus({
+        xvsTokenMultichainContract,
+        chainXvsProxyOftDestContractAddress,
+      });
+    }
+    return undefined;
+  }, [xvsTokenMultichainContract, chainXvsProxyOftDestContractAddress]);
```

Comment:
I recommend staying consistent with other hooks and use the `callOrThrow` function to wrap the query used inside `useQuery`. You'd provide the parameters `xvsTokenMultichainContract` and `chainXvsProxyOftDestContractAddress` and so the hook would throw if one of these does not exist.
I'd then use the `enabled` option of the hook to disable the query when `destinationChainId` corresponds to BSC mainnet or testnet.

This could prevent confusion in the future if someone tried to use this hook and received `undefined` as data, while no error is being thrown and loading indicators (`isLoading`, `isFetching`) are `false`.

Let me know your thoughts.

Reply:
Late reply, but thanks for the suggestion! It fits perfectly with the pattern used all around.